web/middleware/auth.go

```diff
@@ -3,15 +3,48 @@ package middleware
 import (
 	"net/http"
 
+	"github.com/goji/httpauth"
+	"github.com/gorilla/mux"
+
 	"github.com/bonnyci/quartermaster/lib"
 )
 
+var authDefaults = httpauth.AuthOptions{
+	Realm:    "quartermaster",
+	AuthFunc: lib.AuthToken,
+}
+
+func AuthMiddleware(h http.HandlerFunc) http.HandlerFunc {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		httpauth.BasicAuth(authDefaults)(h).ServeHTTP(w, r)
+	})
+}
+
 func AdminMiddleware(h http.HandlerFunc) http.HandlerFunc {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		aU, _, _ := r.BasicAuth()
 		if !lib.IsAdmin(aU) {
+			http.Error(w, "Not member of Admin group", http.StatusUnauthorized)
 			return
 		}
+
 		h.ServeHTTP(w, r)
 	})
 }
+
+func SelfCheckMiddleware(h http.HandlerFunc) http.HandlerFunc {
```

Comment:
I will remove this. I didn't realize I left it. I moved the api endpoints for modifying a status to use the authenticated user instead of passing a user in as we only want a user to modify their own.